server/src/capture.rs

```diff
@@ -0,0 +1,198 @@
+    let config = load_config().expect("Failed to load database configuration");
+
+    let db_user = &config.db_user;
+    let db_password = &config.db_password;
+    let db_name = &config.db_name;
+    let db_host = &config.db_ip;
+
+    info!(
+        "Connecting to Database:[{}] IP:[{}] as User [{}]...",
+        db_name, db_host, db_user
+    );
+
+    // Build the connection string
+    let conn_str = format!(
+        "host={} user={} password={} dbname={}",
+        db_host, db_user, db_password, db_name
+    );
+
+    // Connect to the database
+    let (client, connection) = tokio_postgres::connect(&conn_str, NoTls).await?;
+
+    // Spawn the connection handling in the background
+    tokio::spawn(async move {
+        if let Err(e) = connection.await {
+            error!("Database connection error: [{}]", e);
+        }
+    });
+
+    // Print a message for successful DB connection
+    info!(
+        "Successfully Connected to Database [{}] as User [{}]",
+        db_name, db_user
+    );
+
+    Ok(client)
+}
+
+#[tokio::main]
+pub async fn run() -> std::io::Result<()> {
+    // Initialize logging
+    let log_file_path = "event_capture.log";
+    let log_file = File::create(log_file_path)?; // Clears the log file for every run
+
+    let config = {
+        let mut cfg_builder = ConfigBuilder::new();
+        let cfg_builder = match cfg_builder.set_time_offset_to_local() {
+            Ok(cfg) => cfg,
+            Err(cfg) => cfg, // Proceed even if setting the local time offset fails
+        };
+        cfg_builder
+            //.set_time_format(Some("%Y-%m-%d %H:%M:%S"))
+            .build()
+    };
+
+    CombinedLogger::init(vec![
+        TermLogger::new(
+            LevelFilter::Info,
+            config.clone(),
+            TerminalMode::Mixed,
+            ColorChoice::Auto,
+        ),
+        WriteLogger::new(LevelFilter::Info, config, log_file),
+    ])
+    .unwrap();
+
+    info!("Starting Event Capture Server");
+
+    // Start the TCP server
+    let listen_addr = "0.0.0.0";
+    let listen_port = 3947;
+    let listener = TcpListener::bind((listen_addr, listen_port)).await?;
+    info!("Server listening on {}:{}", listen_addr, listen_port);
+
+    // Create a channel for events
+    let (event_tx, mut event_rx) = mpsc::channel::<Event>(100);
+
+    // Spawn a task to handle database operations
+    tokio::spawn(async move {
+        match db_connect().await {
+            Ok(db_client) => {
+                while let Some(event) = event_rx.recv().await {
+                    if let Err(e) = insert_event(&db_client, &event).await {
+                        error!("Failed to insert event: {}", e);
+                    } else {
+                        info!("Event inserted: {:?}", event);
+                    }
+                }
+            }
+            Err(e) => {
+                error!("Failed to connect to DB: {}", e);
+            }
+        }
+    });
+
+    loop {
```

Comment:
Drop this -- no need to run a TCP server, you can simply directly call the logging code from the webserver.